app/views/songs/_mini_groove_song_card.html.erb

```diff
@@ -0,0 +1,14 @@
+<div class="songcard">
+	<span>
+	<div class="coverart">
+		<a href=<%= url %> >
+		<% unless coverart == '' %>
```

Comment:
It feels like you are checking for empty coverart everywhere - in multiple controllers and in the view.  Let's move the logic around this to the model and possible a helper method.